### PR TITLE
Bugfix/#1270 Fixing Number of project listed on org index includes active AND archived projects 

### PR DESCRIPTION
**Pull request by pamungkaski**
#1270 
### Proposed changes in this pull request

- add test on organization/tests/test_views_default_organizations.py
- fix the bug on organization/views/default.py by filtering project__archived

### When should this PR be merged

- Anytime

### Risks

- Null

### Follow-up actions

- Null

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 
- [ ] **Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 
- [ ] **Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts.

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled. 

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests. 
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/). 

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?** 
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes must be documented in the [Cadasta Platform Documentation](https://github.com/Cadasta/cadasta-docs).
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented in the [API docs](https://github.com/Cadasta/api-docs).
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

**Comment by pamungkaski** [comment]
@oliverroick here it is the new PR,
i'm going to work on the lint now
so sorry and thank you

**Comment by pamungkaski** [comment]
I think the unitest still wrong

**Comment by pamungkaski** [comment]
@oliverroick hi, thanks for your response
I think it is because i start this PR before https://github.com/Cadasta/cadasta-platform/commit/378ea3ab7781a6deccbfa37af43cf1fad6d015f8 merged to the master
and then i tried to update my forked master by pull the parents master
so it counted as commit in my forked master

But, this PR is not done yet i think because the uni-test i make ( https://github.com/Cadasta/cadasta-platform/pull/1357/commits/e8fbd0fddc2be224f7b7422d09de864a716e517c ) is not testing the bug.

I know the test was wrong because @seav say that in the other contributor RP that look like similar to the test that i create.

**Comment by pamungkaski** [comment]
@oliverroick Hi there
I am finish with creating the test, hope you can review it asap

**Comment by seav** [comment]
@pamungkaski, your updated test still does not work. Please try the following:

1. Change cadasta/organization/views/default.py back to how it looks like before the PR.
2. Run your new test. It _should_ fail.
3. Change cadasta/organization/views/default.py to include the proposed solution.
4. Run your new test. It _should_ pass.

If the test does not fail in step 2 and does not pass in step 4, then there is something wrong with your test. Right now, your test will pass at both steps, so your "test" is not actually a test.

**Comment by pamungkaski** [comment]
Hi @seav thanks for your response.
so in the test i should call for the num_projects on the cadasta/organization/views/default.py ?
I'm not sure how to do that because actually my test now is testing the approach that i used to fix the bug by giving statements to count only the project that archived=False.
In this case, fixing the bug only by filtering that.

**Comment by pamungkaski** [comment]
@seav @oliverroick PR updated and i think it is good enough :)

File: cadasta/organization/tests/test_views_default_organizations.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 from ..models import Organization, OrganizationRole, Project, ProjectRole
 from .. import forms
 from .factories import OrganizationFactory, ProjectFactory, clause
+from ..views.default import OrganizationList
```
Comment (seav) [inline cadasta/organization/tests/test_views_default_organizations.py]: We are already importing `views.default` a few lines up so you should be able to reference `OrganizationList` using `default.OrganizationList`.

File: cadasta/organization/views/default.py [hunk 2]
```diff
@@ -37,8 +37,21 @@ class OrganizationList(PermissionRequiredMixin, generic.ListView):
...
+        (Case
+         (When
+          (projects__archived=0, then=1),
+          default=0,
+          output_field=IntegerField()
+          )
+         )
+    )
+
+    def get_qset(self):
```
Comment (seav) [inline cadasta/organization/views/default.py]: Adding a new method that is not used in the platform but is only used for testing is not really a good idea. Just instantiate the view in the test and then access the instance's `queryset` attribute.

File: cadasta/organization/views/default.py [hunk 3]
```diff
@@ -37,8 +37,21 @@ class OrganizationList(PermissionRequiredMixin, generic.ListView):
                                   else ('org.view_archived',))
 
     # This queryset annotation is needed to avoid generating a query for each
-    # organization in order to count the number of projects per org
-    queryset = Organization.objects.annotate(num_projects=Count('projects'))
+    # organization in order to count the number of projects
+    # per org that not archived
+    queryset = Organization.objects.annotate(
+        num_projects=Sum
```
Comment (seav) [inline cadasta/organization/views/default.py]: I suggest placing the opening parenthesis on the same line as the opening construct. And then use 4 spaces for indentation.

File: cadasta/templates/party/party_delete.html [hunk 4]
```diff
@@ -12,11 +12,11 @@
           </button>
           <h3 class="modal-title">{% trans "Delete party" %}</h3>
         </div>
-  
+
         <div class="modal-body">
-          <p>{% trans "Are you sure you want to delete this party?" %}</p>
+          <p>{% trans "Are you sure you want to delete this party? This party currently in ", 8 ,"relationship(s) %}</p>
```
Comment (seav) [inline cadasta/templates/party/party_delete.html]: What is this change? This is not related to the current bug.

File: cadasta/organization/tests/test_views_default_organizations.py [hunk 5]
```diff
@@ -64,6 +65,16 @@ def setup_template_context(self):
             'is_superuser': False,
         }
 
+    def test_should_only_active_projects_added_at_org_index(self):
```
Comment (seav) [inline cadasta/organization/tests/test_views_default_organizations.py]: I don't think this test algorithm is adequate because you are only testing one organization. You should improve the test so that you are checking that `num_projects` is correct for all organizations returned by the queryset, and avoid trivial cases like `num_projects` = 0 because that is the default situation in this test class provided by `setup_models()`.

File: cadasta/templates/party/party_delete.html (hunk 4 above)
Comment (pamungkaski) [inline cadasta/templates/party/party_delete.html]: sorry gonna resolve this

Comment (pamungkaski) [inline cadasta/templates/party/party_delete.html]: forget to change the branch sorry

**Comment by pamungkaski** [comment]
Accidentally closed and cannot be re-opened i don't know why
going to make another PR
